src/PostgREST/Error.hs

```diff
@@ -442,6 +464,10 @@ pgErrorStatus authed (SQL.SessionUsageError (SQL.QueryError _ _ (SQL.ResultError
         "42P01"   -> HTTP.status404 -- undefined table
         "42501"   -> if authed then HTTP.status403 else HTTP.status401 -- insufficient privilege
         'P':'T':n -> fromMaybe HTTP.status500 (HTTP.mkStatus <$> readMaybe n <*> pure m)
+        "PGRST"   ->
+          case (parseMessage m, parseDetails d) of
+            (Just _, Just r) -> HTTP.mkStatus (getStatus r) "CustomHTTPResponse"
```

Comment:
Hm, what does `CustomHTTPResponse` mean in this context?

Reply:
I was thinking since we are can return any HTTP status and headers through this interface, so just called it `CustomHTTPResponse` without thinking too much. How about `PgRaiseResponse` or maybe just leave it blank `""`?

Reply:
@steve-chavez Please review and let me know.

Reply:
This should be using the default HTTP status messages, e.g. if the user specifies `{"status": 404,...}", then the message should be "Not Found". It can be achieved by doing:

```haskell
(Just _, Just r) -> toEnum (getStatus r)
```

But maybe we could also add further extensibility by letting the user specify the status description if they wish. For instance:

```json
{"status": 332, "description": "My Custom Status", ...}
```

Then you can do something like (didn't test this, just an idea):

```haskell
(Just _, Just r) -> maybe  (toEnum $ getStatus r) (HTTP.mkStatus $ getStatus r) (getDescription r)
```

For that, the `PgRaiseErrDetails` needs another field like:

```haskell
getDescription :: Maybe ByteString
```

Something like that (Maybe ByteString is not allowed, use Text instead).